shared/common-adapters/name-with-icon/index.js

```diff
@@ -0,0 +1,227 @@
+// @flow
+import * as React from 'react'
+import * as Styles from '../../styles'
+import Avatar from '../avatar'
+import Box from '../box'
+import ClickableBox from '../clickable-box'
+import Icon, {castPlatformStyles, type IconType} from '../icon'
+import Text, {type TextType} from '../text'
+import ConnectedUsernames from '../usernames/container'
+
+type Size = 'small' | 'default' | 'large'
+
+// Exposed style props for the top-level container and box around metadata arbitrarily
+export type NameWithIconProps = {|
+  avatarStyle?: Styles.StylesCrossPlatform,
+  colorFollowing?: boolean,
+  containerStyle?: Styles.StylesCrossPlatform,
+  editableIcon?: boolean,
+  horizontal?: boolean,
+  icon?: IconType,
+  isYou?: boolean,
+  metaOne?: string | React.Node,
+  metaStyle?: Styles.StylesCrossPlatform,
+  metaTwo?: string | React.Node,
+  onClick?: (SyntheticEvent<> | void) => void,
```

Comment:
Can `(SyntheticEvent<> | void)` just be `()`?

Reply:
originally, the type was `any => void`. Everywhere we use an onClick on NameWithIcon seems to just be of the type `() => void`, except for in one story, where we do some [interesting stuff with the event handler](https://github.com/keybase/client/blob/b59f99671eace7cf47673c87a505e0a350000983/shared/common-adapters/name-with-icon/index.stories.js#L15). Hence the `(SyntheticEvent<> | void) => void` type.

I think we could change the type to `() => void` since I can't really think of a good situation that we would need to distinguish outer and inner clicks, but maybe @buoyad could better say since he wrote the story?